lib/tests/unit/modules/download_file_test.cc

```diff
@@ -58,17 +60,68 @@ static const std::string CRT { "mock_crt" };
+                                                                            "}"
+                                                                    "},"
+                                                                    "\"sha256\":\"94CBA5396781C06EFB4237730751532CBEFEA4C637D17A61B2E78598F08732C2\","
+                                                                    "\"destination\":\"%1%/file.txt\","
+                                                                    "\"link_source\":\"\","
+                                                                    "\"file_type\":\"file\""
+                                                                "},"
+                                                                "{"
+                                                                    "\"uri\":{"
+                                                                            "\"path\":\"\","
+                                                                            "\"params\":{}"
+                                                                    "},"
+                                                                    "\"sha256\":\"\","
+                                                                    "\"destination\":\"%2%\","
+                                                                    "\"link_source\":\"\","
+                                                                    "\"file_type\":\"directory\""
+                                                                "}"
+                                                            "]"
+                                                "}") % TEST_FILE_DIR % TEST_NEW_DIR;
+
+// Create a failure scenario by using a request with a "directory" file type where
+// the destination already exists as a file
+static auto dir_already_exists_params = boost::format("{\"files\": ["
+                                                                "{"
+                                                                    "\"uri\":{"
+                                                                            "\"path\":\"\","
+                                                                            "\"params\":{}"
+                                                                    "},"
+                                                                    "\"sha256\":\"\","
+                                                                    "\"destination\":\"%1%/file.txt\","
+                                                                    "\"link_source\":\"\","
+                                                                    "\"file_type\":\"directory\""
+                                                                "}"
+                                                            "]"
+                                                       "}") % TEST_FILE_DIR;
+
+// Create a failure scenario by using a request with a "directory" file type where
```

Comment:
directory file type?

Reply:
Yeah, the field in the schema I used to identify if it was a file/dir/link is called "file_type". I can change that.